File: py/src/sols/p13.py
# https://projecteuler.net/problem=13
# --------------------------------------------------------------------------------------
# Find the first 10 digits of the sum of all the 50-digit numbers.

from __future__ import annotations

from src.utils import parse_file
from src.utils.files import PROBLEM_13


def solve() -> None:
    big_sum: int = 0
    big_nums: list[int] = [int(x) for x in parse_file(PROBLEM_13, ",")]

    for num in big_nums:
        big_sum += num

    print(
        f"First 10 digits of the sum of all given 50-digit numbers: {str(big_sum)[:10]}"
    )

File: rs/src/sols/p13.rs
// https://projecteuler.net/problem=13
// -------------------------------------------------------------------------------------
// Find the first 10 digits of the sum of all the 50-digit numbers.

use std::time::Instant;

use crate::utils::stopwatch;

#[inline]
pub fn solve_p13() {
    const NUMS: &str = include_str!("../../../inputs/13-numbers.txt");

    let start = Instant::now();

    let answer = NUMS
        .split(',')
        .filter_map(|s| s[0..12].parse::<u64>().ok())
        .sum::<u64>()
        .to_string()
        .get(0..10)
        .expect("result shouldn't be empty!")
        .to_string();

    let duration = start.elapsed();

    println!("{answer}");
    stopwatch(duration);
}
